File: solutions/scala/8 kyu/HowMuchWaterDoINeed.scala
//https://www.codewars.com/kata/575fa9afee048b293e000287/
object Sol {
  def howMuchWater(water: Int, maxLoad: Int, clothes: Int): Either[Double, String] = {
    if (clothes > maxLoad * 2) {
      Right("Too much clothes")
    } else if (clothes < maxLoad) {
      Right("Not enough clothes")
    } else {
      val waterUsed = water * math.pow(1.1, clothes - maxLoad)
      Left(f"$waterUsed%1.2f".toDouble)
    }
  }
}

File: solutions/py/8 kyu/HowMuchWaterDoINeed.py
#!/usr/bin/python3

# https://www.codewars.com/kata/575fa9afee048b293e000287/ (btw the single worst kata on codewars)
def how_much_water(water, load, clothes):
    if clothes > 2 * load:
        return "Too much clothes"
    elif clothes < load:
        return "Not enough clothes"
    else:
        return round(water * 1.1 ** (clothes - load), 2)
